src/encoded/static/components/app.js

```diff
@@ -69,277 +87,967 @@ const biosampleTypeList = [
     'whole organisms',
     'stem cell',
     'in vitro differentiated cells',
-    'induced pluripotent stem cell line'
+    'induced pluripotent stem cell line',
 ];
 
 
-// See https://github.com/facebook/react/issues/2323
-var Title = React.createClass({
-    render: function() {
-        return <title {...this.props}>{this.props.children}</title>;
-    },
-    componentDidMount: function() {
-        var node = document.querySelector('title');
-        if (node && !node.getAttribute('data-reactid')) {
-            node.setAttribute('data-reactid', this._rootNodeID);
+// See https://github.com/facebook/react/issues/2323 for an IE8 fix removed for Redmine #4755.
+const Title = props => <title {...props}>{props.children}</title>;
+
+Title.propTypes = {
+    children: PropTypes.node.isRequired,
+};
+
+
+// Get the current browser cookie from the DOM.
+function extractSessionCookie() {
+    const cookie = require('cookie-monster');
+    return cookie(document).get('session');
+}
+
+
+function contentTypeIsJSON(contentType) {
+    return (contentType || '').split(';')[0].split('/').pop().split('+').pop() === 'json';
+}
+
+
+// Extract the current session information from the current browser cookie.
+function parseSessionCookie(sessionCookie) {
+    const buffer = require('buffer').Buffer;
+    let session;
+    if (sessionCookie) {
+        // URL-safe base64
+        const mutatedSessionCookie = sessionCookie.replace(/-/g, '+').replace(/_/g, '/');
+        // First 64 chars is the sha-512 server signature
+        // Payload is [accessed, created, data]
+        try {
+            session = JSON.parse(buffer(mutatedSessionCookie, 'base64').slice(64).toString())[2];
+        } catch (e) {
+            console.warn('session parse err %o', session);
         }
     }
-});
+    return session || {};
+}
+
+
+function recordServerStats(serverStats, timingVar) {
```

Comment:
recordServerStats is in the mixins.js file in master. Had to move it since I removed mixins.js.